### PR TITLE
[14.0][MIG] report_csv: Migration to 14.0

### PR DESCRIPTION
**Pull request by ferran-S73**
Original pull request https://github.com/OCA/reporting-engine/pull/442

Updates and resolves conflicts

### CODE REVIEW

File: report_csv/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Base report csv",
+    "summary": "Base module to create csv report",
+    "author": "Creu Blanca, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/reporting-engine",
+    "category": "Reporting",
+    "version": "14.0.1.0.0",
+    "license": "AGPL-3",
+    "external_dependencies": {"python": ["python-csv"]},
```
Comment (davejames) [inline report_csv/__manifest__.py]: Is there a need for this dependency? This module works well just using the regular csv module which is already available in python.

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # generated from manifests external_dependencies
 PyPDF2
+python-csv
```
Comment (davejames) [inline requirements.txt]: also here

**Review by davejames — CHANGES_REQUESTED** [review]

File: report_csv/__manifest__.py (hunk 1 above)
Comment (ferran-S73) [reply report_csv/__manifest__.py]: I think you're right, I'll push an update removing this dependency

**Review by newtratip — APPROVED** [review]

**Review by ioans73 — APPROVED** [review]

**Comment by ferran-S73** [comment]
@davejames could you review please?

**Review by Reyes4711-S73 — APPROVED** [review]

**Review by andreampiovesana — APPROVED** [review]
merge?

**Review by adrienpeiffer — APPROVED** [review]

**Comment by adrienpeiffer** [comment]
@OCA/reporting-engine-maintainers Could we merge this ?

**Review by etobella — APPROVED** [review]

**Comment by etobella** [comment]
Can you rebase?

**Comment by ferran-S73** [comment]
> Can you rebase?

@etobella done

**Comment by etobella** [comment]
tests fail might be unrelated, but can you reexecute pre-commit?

**Comment by ferran-S73** [comment]
> tests fail might be unrelated, but can you reexecute pre-commit?

It's working just fine on my local env

**Comment by etobella** [comment]
Setup folder has not been added

**Comment by ferran-S73** [comment]
> Setup folder has not been added

I've added the setup folder generated by pre-commit locally

**Comment by ferran-S73** [comment]
@etobella @adrienpeiffer @andreampiovesana @newtratip @OCA/reporting-engine-maintainers anyone has idea why is the test failing to donwload openjdk?

**Comment by etobella** [comment]
There was an error with the CI Image. Now it has been fixed :smile:

**Pull request merged**